x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/dataframe/DataFrameAnalyticsConfigUpdate.java

```diff
@@ -152,6 +154,23 @@ public boolean requiresRestart(DataFrameAnalyticsConfig source) {
             || (getMaxNumThreads() != null && getMaxNumThreads().equals(source.getMaxNumThreads()) == false);
     }
 
+    public Set<String> getUpdatedFields() {
```

Comment:
A `Set` here seems like overkill. Just a `List` and `ArrayList` would do. 

But, that is not that big of a deal.

Reply:
I'm using a `TreeSet` and thus I like it because we get the fields sorted no matter which way they are added in the future.